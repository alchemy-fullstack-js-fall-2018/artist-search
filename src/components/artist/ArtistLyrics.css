.lyric {
  font-size: 40px;
}
