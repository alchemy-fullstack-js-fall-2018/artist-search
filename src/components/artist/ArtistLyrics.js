import React, { Component } from 'react';
import { getLyrics } from '../../services/musicSearch';
import styles from './ArtistLyrics.css';

export default class ArtistLyrics extends Component {
  state = {
    lyrics: '',
    artist: '',
    title: ''
  }

  componentDidMount() {
    const artist = this.props.match.params.artistName;
    const title = this.props.match.params.songName;

      getLyrics( artist, title ).then(result => {
      this.setState({ lyrics: result.lyrics, artist, title })
    });
  }

  render() {
    const { title, artist, lyrics } = this.state;

    return (
      <div className={styles.lyric}>
        <h3>{artist}</h3>
        <h4>{title}</h4>
        <p>{lyrics}</p>
      </div>
    );
  }
}
