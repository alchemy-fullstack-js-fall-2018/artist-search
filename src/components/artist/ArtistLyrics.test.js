import React from 'react';
import { shallow } from 'enzyme';
import ArtistLyrics from './ArtistLyrics';

jest.mock('../../services/musicSearch', () => {
  return { getLyrics: () => Promise.resolve({ lyrics: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quod quaerat quae quam iusto omnis reprehenderit amet repellat. Atque accusantium suscipit rem sint quisquam repellat. Enim fuga excepturi magni blanditiis officia.' }) };
});

describe('Lyrics test', () => {

  let wrapper;
  const match = { params: {artistName: 'Mariah Carey', songName: 'Butterflies'} };

  beforeEach(() => {
    wrapper = shallow(<ArtistLyrics match={match}/>);
  });

   it('renders an Lyrics Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
