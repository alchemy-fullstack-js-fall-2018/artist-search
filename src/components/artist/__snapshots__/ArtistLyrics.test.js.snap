// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics test renders an Lyrics Component 1`] = `
<div
  className="lyric"
>
  <h3>
    Mariah Carey
  </h3>
  <h4>
    Butterflies
  </h4>
  <p>
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quod quaerat quae quam iusto omnis reprehenderit amet repellat. Atque accusantium suscipit rem sint quisquam repellat. Enim fuga excepturi magni blanditiis officia.
  </p>
</div>
`;
